internal/beater/config/agentconfig_fips.go

```diff
@@ -0,0 +1,24 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build requirefips
+
+package config
+
+func generateEtag(s *FleetAgentConfig) (string, error) {
+	return "", nil
```

Comment:
Looking at this functionality - why not generating an `etag` with a FIPS approved algorithm rather than returning an empty string? As @carsonip pointed out recently, the `etag` is used for caching on the apm agent side.